models/classes/user/KvLtiUserService.php

```diff
@@ -92,7 +103,17 @@ public function getUserIdentifier($ltiUserId, $consumer)
      */
     public function getUserDataFromId($taoUserId)
     {
-        $data = $this->getPersistence()->get($taoUserId);
+        if (!$this->getPersistence()->exists(self::LTI_USER_LOOKUP . $taoUserId)) {
+            if ($this->getPersistence()->exists($taoUserId)) {
+                $this->getPersistence()->set(self::LTI_USER_LOOKUP . $taoUserId, $taoUserId);
+                $data = $this->getPersistence()->get($taoUserId);
+            } else {
+                return false;
```

Comment:
You are breaking the method definition by returning null, in the abstract it's saying that it's an array, and in the `OntologyLtiUserService` it's returning  null, can you return null as well.

Reply:
hum, `You are breaking the method definition by returning null` versus `can you return null as well`